api/controllers/UserController.js

```diff
@@ -50,33 +51,37 @@ module.exports = {
       return res.badRequest(err);
     }
   },
+
   reLogin: function (req, res) {
     userAPI
       .findUserByUsername(req.body.username)
       .then(function gotUser(user) {
+        const { game: gameId, loggedIn } = req.session;
         const checkPass =
-          !req.session.loggedIn && req.body.password
+          !loggedIn && req.body.password
             ? passwordAPI.checkPass(req.body.password, user.encryptedPassword)
             : null;
-        const promiseGame = gameService.populateGame({ gameId: user.game });
+        const promiseGame = gameId ? gameService.findGame({ gameId }) : null;
         return Promise.all([promiseGame, Promise.resolve(user), checkPass]);
       })
       .then((values) => {
         const [ game, user ] = values;
         req.session.loggedIn = true;
         req.session.usr = user.id;
-        req.session.game = game.id;
         req.session.pNum = user.pNum;
-        Game.subscribe(req, [game.id]);
         sails.sockets.join(req, 'GameList');
-        Game.publish([game.id], {
-          verb: 'updated',
-          data: {
-            ...game.lastEvent,
-            game,
-          },
-        });
-
+        // Set session data & send socket msg if in game
+        if (game) {
```

Comment:
This seems new, and now it's behind a check for game's existence. Do we ever expect this to not be set at this point?

Reply:
So I may need to roll back on this but the usecase is for switching sessions between "other" users during testing, which we do in the home.spec while both those "other" users are not already in the game so they can ready up etc.

This is a new change as a part of removing my previous implementation of session switching (which used a new testing-only endpoint) in favor of reusing this existing one.

That change broke a reconnection spec so it may need to roll back. That's the CI failure and I need to debug still

Reply:
Fixed by reworking UserController.reLogin: https://github.com/cuttle-cards/cuttle/pull/345/commits/91fa83a63ac42b99197bc8d029b29929c52c8508